File: my-shop-selenium/src/main/java/data/factory/UserDatafactory.java
package data.factory;

import data.dto.UserDTO;
import net.datafaker.Faker;
import util.Enviroment;

public class UserDatafactory {

    private static final Faker faker = new Faker();

    public static UserDTO getFakerUser() {
        return new UserDTO(
            faker.internet().emailAddress(),
            faker.internet().password(),
            faker.name().firstName(),
            faker.name().lastName(),
            String.valueOf(faker.number().numberBetween(1, 28)),
            String.valueOf(faker.number().numberBetween(1, 12)),
            String.valueOf(faker.number().numberBetween(1900, 2021)),
            faker.bool().bool()
        );
    }

    public static UserDTO getDefaultUser() {
        return new UserDTO(
            Enviroment.getMyshopDefaultEmail(),
            Enviroment.getMyshopDefaultPassword(),
            "Fulano",
            "de Tal",
            "15",
            "06",
            "1990",
            true
        );
    }

    public static UserDTO getDefaultUserToUpdatePersonalInformation() {
        UserDTO myUser = getDefaultUser();
        myUser.setFistName("Goularte");
        myUser.setLastName("Goncalves");
        myUser.setDayOfBirth("16");
        myUser.setMonthOfBirth("07");
        myUser.setYearOfBirth("1991");
        myUser.setMen(false);
        return myUser;
    }

    public static UserDTO getDefaultUserToUpdateEmail() {
        UserDTO myUser = getDefaultUser();
        myUser.setEmail("second.emal@email.com");
        return myUser;
    }

    public static UserDTO getDefaultUserToUpdatePassword() {
        UserDTO myUser = getDefaultUser();
        myUser.setPassword("newpassword");
        return myUser;
    }

}

File: my-shop-cypress/cypress/e2e/data/factory/user-datafactory.ts
import UserDTO from '../dto/user-dto';
import { faker } from '@faker-js/faker';


export default class UserDatafactory {
    static getFakerUser(): UserDTO {
        return new UserDTO(
            faker.internet.email(),
            faker.internet.password(),
            faker.person.firstName(),
            faker.person.lastName(),
            String(faker.number.int({ min: 1, max: 28 })),
            String(faker.number.int({ min: 1, max: 12 })),
            String(faker.number.int({ min: 1900, max: 2021 })),
            faker.datatype.boolean()
        );
    }

    static getDefaultUser(): UserDTO {
        return new UserDTO(
            Cypress.env("MYSHOP_DEFAUL_EMAIL"),
            Cypress.env("MYSHOP_DEFAUL_PASSWORD"),
            'Fulano',
            'de Tal',
            '15',
            '6',
            '1990',
            true
        );
    }

    static getDefaultUserToUpdatePersonalInformation(): UserDTO {
        const myUser = UserDatafactory.getDefaultUser();
        myUser.fistName = 'Goularte';
        myUser.lastName = 'Goncalves';
        myUser.dayOfBirth = '16';
        myUser.monthOfBirth = '7';
        myUser.yearOfBirth = '1991';
        myUser.isMen = false;
        return myUser;
    }

    static getDefaultUserToUpdateEmail(): UserDTO {
        const myUser = UserDatafactory.getDefaultUser();
        myUser.email = 'second.email@email.com';
        return myUser;
    }

    static getDefaultUserToUpdatePassword(): UserDTO {
        const myUser = UserDatafactory.getDefaultUser();
        myUser.password = 'newpassword';
        return myUser;
    }
}
